src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,257 @@
+
+                if (symbol.IsGenericMethod)
+                {
+                    var enclosing = enclosingTypeParameters.Select(p => (TypeSyntax)p.Name.ToIdentifierName());
+                    var typeArguments = enclosing.Concat(symbol.TypeArguments.Select(x => x.GenerateTypeSyntax()));
+                    updatedInvocation = updatedInvocation.WithExpression(
+                        (ExpressionSyntax)generator.WithTypeArguments(updatedInvocation.Expression, typeArguments));
+                }
+
+                updatedInvocation = updatedInvocation.WithExpression(WithName(updatedInvocation.Expression, methodName));
+
+                editor.ReplaceNode(currentInvocation, updatedInvocation);
+            }
+
+            return document.WithSyntaxRoot(editor.GetChangedRoot());
+        }
+
+        private static SyntaxNode WithTypeConstraint(SyntaxNode node, ITypeParameterSymbol sym, SyntaxGenerator g)
+        {
+            var typeConstraintKind = SpecialTypeConstraintKind.None;
+            if (sym.HasConstructorConstraint)
+            {
+                typeConstraintKind |= SpecialTypeConstraintKind.ReferenceType;
+            }
+
+            if (sym.HasValueTypeConstraint)
+            {
+                typeConstraintKind |= SpecialTypeConstraintKind.ValueType;
+            }
+
+            if (sym.HasConstructorConstraint)
+            {
+                typeConstraintKind |= SpecialTypeConstraintKind.Constructor;
+            }
+
+            return g.WithTypeConstraint(node, sym.Name, typeConstraintKind, sym.ConstraintTypes.Select(x => x.GenerateTypeSyntax()));
+        }
+
+        private static ExpressionSyntax WithName(ExpressionSyntax expression, SyntaxToken name)
+        {
+            switch (expression.Kind())
+            {
+                case SyntaxKind.IdentifierName:
+                    var sname = (SimpleNameSyntax)expression;
+                    return sname.WithIdentifier(name);
+
+                case SyntaxKind.GenericName:
+                    var gname = (GenericNameSyntax)expression;
+                    return gname.WithIdentifier(name);
+
+                default:
+                    return expression;
+            }
+        }
+
+        private static bool SupportsNonTrailingNamedArguments(ParseOptions options)
+            => ((CSharpParseOptions)options).LanguageVersion >= LanguageVersion.CSharp7_2;
+
+        private static List<ISymbol> GetCapturedTypeParameters(ISymbol symbol)
+        {
+            var results = new List<ISymbol>();
+
+            while (symbol != null)
+            {
+                results.AddRange(symbol.GetTypeParameters());
+                symbol = symbol.ContainingSymbol;
+                if (symbol.Kind == SymbolKind.NamedType)
+                {
+                    break;
+                }
+            }
+
+            return results;
+        }
+
+        private static SyntaxTokenList GetModifiers(bool isStatic)
```

Comment:
i'd much rather you just make an IMethodSymbol and use the existing codepaths we have for generating source from a symbol.